internal/server/grpc/grpc.go

```diff
@@ -179,3 +207,68 @@ func loadX509KeyPair(certfile, keyfile string) (tls.Certificate, error) {
 	}
 	return tls.X509KeyPair(cert, key)
 }
+
+func generateCertificatesForATLS(svc *agent.Service) ([]byte, []byte, error) {
```

Comment:
since agent has the ability to receive certs for tls, I would say we would rather parse those certs and add extensions to the certs once read

Reply:
I think we can avoid this. By adding the hash of the public key in the `report data` field of the attestation report. This is the the public key of the generated certificate (self-signed) for TLS. The idea is to have the attestation report be a part of the x.509 certificate (as an extension). Because the attestation report is signed by AMD, we can verify that the x.509 certificate is generated inside the enclave. In this way, we can avoid using server certificates from the manager and the clients can be sure that the key used in TLS communication is created in the SVM.

In order to verify that the SVM is used for a certain computation, we can hash the computation and have it be a part of the attestation report by using the `host data` field of the attestation report. This hash is supplied to the qemu command when we boot the SVM (there is a parameter `-host_data` in the qemu). This way the Agent can verify the computation (manifest) that it is getting from the manager through vsock. 

To verify that the SVM is running on the certain backend, we can use the `block ID` structure that is fed to qemu and also a part of the attestation report of SVM. This structure contatins hash of the key of the owner that started the SVM. This key can be a key of the Manager.

Reply:
ok, although I suspect on the client/cli we'll need to skip verify since the self signed certs will have an unknown CA

Reply:
Yes, but the certificate is checked if it is self-signed.